Sources/Basic/Result.swift

```diff
@@ -161,7 +161,18 @@ extension Result where ErrorType == StringError {
     }
 }
 
-extension Result: Equatable where Value: Equatable, ErrorType: Equatable {}
+extension Result: Equatable where Value: Equatable, ErrorType: Equatable {
```

Comment:
Why do we need to do this? Does the synthesized implementation not work properly?

Reply:
The project is not compilable for me on Xcode 9.4.1/Xcode 10b6. So I fixed the compilation error by providing this implementation. I wonder how does my setup differ from your CI as I suppose the build works fine on your machines.

Reply:
Please use the latest Xcode beta with a trunk snapshot, see: https://github.com/apple/swift-package-manager/blob/master/Documentation/Development.md#using-trunk-snapshot